src/main/java/org/betonquest/betonquest/events/DropItemEvent.java

```diff
@@ -0,0 +1,166 @@
+import org.bukkit.event.entity.ItemMergeEvent;
+import org.bukkit.event.player.PlayerJoinEvent;
+import org.bukkit.inventory.ItemStack;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.UUID;
+
+@SuppressWarnings({"PMD.CommentRequired", "PMD.FieldNamingConventions", "PMD.UnusedNullCheckInEquals", "PMD.CollapsibleIfStatements"})
+public class DropItemEvent extends QuestEvent implements Listener {
+
+    private final Instruction.Item[] questItems;
+    private final CompoundLocation location;
+    private final boolean isPrivate;
+    private final boolean isIndestructible;
+
+    private final Map<Entity, UUID> entityPlayerMap = new HashMap<>();
+    private final List<Entity> IndestructibleItem = new ArrayList<>();
+    private final EntityHider hider = new EntityHider(BetonQuest.getInstance(), EntityHider.Policy.BLACKLIST);
+
+    public DropItemEvent(final Instruction instruction) throws InstructionParseException {
+        super(instruction, true);
+
+        Bukkit.getPluginManager().registerEvents(BukkitEvent(), BetonQuest.getInstance());
+
+        questItems = instruction.getItemList();
+        location = instruction.getLocation();
+        isPrivate = instruction.hasArgument("private");
+        isIndestructible = instruction.hasArgument("indestructible");
+
+        if (isPrivate) {
+            if (Bukkit.getPluginManager().getPlugin("ProtocolLib") == null) {
+                throw new InstructionParseException("You Need ProtocolLib installed to Use Private Drop Item");
+            }
+        }
+    }
+
+    @Override
+    protected Void execute(final String playerID) throws QuestRuntimeException {
+        for (final Instruction.Item item : questItems) {
+            final QuestItem questItem = item.getItem();
+            final VariableNumber amount = item.getAmount();
+
+            final ItemStack generateItem = questItem.generate(amount.getInt(playerID), playerID);
+            final Player player = PlayerConverter.getPlayer(playerID);
+            final Location loc = location.getLocation(playerID);
+            final Entity droppedItem = loc.getWorld().dropItem(loc, generateItem);
+
+            if (isPrivate) {
+                entityPlayerMap.put(droppedItem, player.getUniqueId());
+                for (final Player onlinePlayer : Bukkit.getOnlinePlayers()) {
+                    if (!onlinePlayer.equals(player)) {
+                        hider.hideEntity(onlinePlayer, droppedItem);
+                    }
+                }
+            }
+            if (isIndestructible) {
+                IndestructibleItem.add(droppedItem);
+            }
+        }
+        return null;
+    }
+
+    @SuppressWarnings({"PMD.MethodNamingConventions", "PMD.NPathComplexity", "PMD.AvoidDuplicateLiterals"})
+    private Listener BukkitEvent() {
+        return new Listener() {
+
+            @SuppressWarnings("PMD.ConfusingTernary")
```

Comment:
You should also not suppress ConfusingTernary

Reply:
But it will send a null exception if I remove check if it null